agent/websockets/connection.go

```diff
@@ -68,7 +68,6 @@ var stripHeaderNames = map[string]bool{
 	"Sec-Websocket-Key":        true,
 	"Sec-Websocket-Version":    true,
 	"Sec-Websocket-Extensions": true,
-	"Sec-Websocket-Protocol":   true,
```

Comment:
I *think* this will result in an error being returned by Gorilla's websocket package when it tries to create the connection to the backend server.

Instead, it looks like we have to take any passed in subprotocols and set them in [the `Subprotocols` field of the Dialer](https://github.com/gorilla/websocket/blob/c3dd95aea9779669bb3daafbd84ee0530c8ce1c1/client.go#L90).

If we just passed in the header as is, then we would hit [this error](https://github.com/gorilla/websocket/blob/c3dd95aea9779669bb3daafbd84ee0530c8ce1c1/client.go#L219).

Reply:
It looks like I was mistaken; the client code only errors out if *both* the header and subprotocols fields are present, so this change is actually good as-is.